js/ad_manager_vast.js

```diff
@@ -483,6 +546,87 @@ OO.Ads.manager(function(_, $) {
     this.initialize = function(amc) {
       this.amc = amc;
       this.amc.addPlayerListener(this.amc.EVENTS.INITIAL_PLAY_REQUESTED, _.bind(this.initialPlay, this));
+      this.amc.addPlayerListener(this.amc.EVENTS.FULLSCREEN_CHANGED, _.bind(_onFullscreenChanged, this));
+      this.amc.addPlayerListener(this.amc.EVENTS.SIZE_CHANGED, _onSizeChanged);
+    };
+
+    /**
+     * Callback when the media file is loaded. Once is loaded we can initialize the ad
+     * This is only required for VPAID ads
+     * @private
+     */
+    var _initializeAd = _.bind(function() {
+      var eventName,
+          environmentVariables,
+          viewMode,
+          creativeData = {};
+
+      currentAd.data = currentAd.ad.data;
+
+      if (typeof iframe.contentWindow.getVPAIDAd !== 'function') {
+        OO.log('VPaid: Required function getVPAIDAd() is not defined.');
+        return;
+      }
+      currentAd.vpaidAd = iframe.contentWindow.getVPAIDAd();
+
+      // Subscribe to ad unit events
+      for (eventName in VPAID_EVENTS) {
+        currentAd.vpaidAd.subscribe(_.bind(_onAdEvent, this, VPAID_EVENTS[eventName]),
+            VPAID_EVENTS[eventName], this);
+      }
+
+      this._slot = _createUniqueElement();
+      this._videoSlot = this.amc.ui.adVideoElement[0];
+
+      environmentVariables = _.extend({
+        slot: this._slot,
+        videoSlot: this._videoSlot,
+        videoSlotCanAutoPlay: true
+      }, this.environmentVariables);
+
+      this._properties = {
+        adWidth: this._slot.offsetWidth,
+        adHeight: this._slot.offsetHeight,
+        adDesiredBitrate: 600,
+      };
+
+      viewMode = _getFsState() ? 'fullscreen' : 'normal';
+      creativeData = {
+        AdParameters: currentAd.ad.adParams
+      };
+
+      this.initAd(this._properties['adWidth'], this._properties['adHeight'], viewMode,
+                  this._properties['adDesiredBitrate'], environmentVariables, creativeData);
+    }, this);
+
+    /**
+     * Initializes the ad by sending the data to the ad unit.
+     * @public
+     * This is only required for VPAID ads
+     * @method Vast#initAd
+     * @param {int} width Width of the slot where the ad will be placed
+     * @param {int} height Height of the slot where the ad will be placed
+     * @param {string} viewMode Can be either `normal` or `fullscreen`
+     * @param {int} desiredBitrate The bitrate for the ad
+     * @param {object} creativeData Contains the aditional ad parameters for the ad
+     * @param {object} environmentVars Contains the slot and videoSlot elements
+     */
+    this.initAd = function(width, height, viewMode, desiredBitrate, creativeData, environmentVars) {
```

Comment:
Can you change this to initVpaidAd?  Please fix this everywhere else so it's clear, for code and documentation.